src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1136,6 +1136,20 @@ IBackendMessage ParseServerMessage(NpgsqlReadBuffer buf, BackendMessageCode code
         internal T ReadExpecting<T>() where T : class, IBackendMessage
             => ReadExpecting<T>(false).Result;
 
+        internal async ValueTask<T> DiscardBackendMessagesUntil<T>(bool async) where T : class, IBackendMessage
+        {
+            T asExpected;
+            do
+            {
+                var msg = await ReadMessage(async);
+                asExpected = msg as T;
+            } while (asExpected == null);
+            return asExpected;
+        }
+
+        internal T DiscardBackendMessagesUntil<T>() where T : class, IBackendMessage
+            => DiscardBackendMessagesUntil<T>(false).Result;
```

Comment:
Prefer `.GetAwait().GetResult()` to `.Result` on tasks, since if the task fails the former throws the exception whereas the latter throws AggregateException, IIRC.

Reply:
Ok. Obsolete if I use `NpgsqlConnector.SkipUntil()` but the style is copied from `NpgsqlConnector.ReadExpecting()` so you might wan to have a look for other places where this is used.

Reply:
Thanks, corrected that.